File: backend/src/main/java/com/grouply/backend/post/post/dto/PostDTO.java
package com.grouply.backend.post.post.dto;

import com.grouply.backend.project.project.Dtos.ProjectDTO;
import com.grouply.backend.post.project_post_position.dto.ProjectPostPositionDTO;
import lombok.Builder;
import lombok.Data;

import java.time.LocalDateTime;
import java.util.List;

@Data
@Builder
public class PostDTO {

    private Long id;
    private String title;
    private String description;
    private List<ProjectPostPositionDTO> positions;
    private ProjectDTO projectDTO;
    private LocalDateTime postedAt;

}

File: frontend/src/features/feed/shared/models/PostDto.ts
import type { ProjectDTO } from "../../posts/models/ProjectDto"
import type { ProjectPostPositionDTO } from "./ProjectPostPositionDto"


export interface PostDTO {
  id: number
  title: string
  description: string
  projectDTO: ProjectDTO
  positions: ProjectPostPositionDTO[]
  postedAt: Date

}
